docker/musicbrainz-server-base/DBDefs.pm.m4

```diff
@@ -0,0 +1,128 @@
+include(`config.all.m4')dnl
+define(`_boolean_def', `subroutine'(``$1'', `ifelse'($1, ``1'', ``1'', ``0'')))dnl
+define(`_code_def', `subroutine'(``$1'', $1))dnl
+define(`_string_def', `subroutine'(``$1'', `quoted_string'(``$1'')))dnl
+define(`_wrap_def', `ifdef'(``$1'', ``$2''(```$1'''), ``dnl''))dnl
```

Comment:
Quotes are kind of hard to explain, but they're the way they are because m4 has no escape character, so you can't just escape quotes. https://www.gnu.org/software/m4/manual/m4-1.4.15/html_node/Quoted-strings.html#Quoted-strings

The quotes always have to be balanced, and the actual value has "one level of quotes stripped off". So `` `broken`macro'name `` has the value `brokenmacroname` and ``` ``broken``macro''name'' ``` has the value `` `broken`macro'name' ``.

The quoting rule of thumb mentioned at https://www.gnu.org/software/m4/manual/m4-1.4.15/html_node/Quoting-Arguments.html is useful for making sense of things.

You don't have to quote parentheses in macro definitions as long as they're balanced. Otherwise you might accidentally close the `define(`.